File: backend/Domain/Entities/Task.cs
using Domain.Common;

namespace Domain.Entities
{
    public class Task : BaseAuditableEntity
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public bool IsCompleted { get; set; }
        public int TaskListId { get; set; }
        public TaskList TaskList { get; set; } = null!;
        public int? UserId { get; set; }
        public User? User { get; set; }
        public int Rank { get; set; } = 1; 
        public string Color { get; set; } = "Default"; 
        public bool IsFavorite { get; set; } = false;
    }
}

File: frontend/src/types/task.ts
import { Priority, Color } from '../utils/taskUtils';

export interface Task {
  id: number;
  title: string;
  description: string;
  rank: number;
  color: Color;
  isCompleted: boolean;
  isFavorite: boolean;
  taskListId: number;
  createdBy?: string | null;
  lastModified?: string | null;
  lastModifiedBy?: string | null;
}

export interface NewTask {
  title: string;
  description: string;
  rank: Priority;
  color: Color;
}

export type UpdateTaskDetails = Partial<Task>;

export interface TaskList {
  id: number;
  name: string;
  description: string;
  createdBy?: string | null;
}

export interface NewTaskList {
  name: string;
  description: string;
}